client-spark/spark-2/src/main/java/org/apache/spark/shuffle/celeborn/HashBasedShuffleWriter.java

```diff
@@ -319,6 +317,15 @@ private void flushSendBuffer(int partitionId, byte[] buffer, int size)
     writeMetrics.incWriteTime(System.nanoTime() - start);
   }
 
+  private void abort() throws IOException {
+    try {
+      dataPusher.waitOnTermination();
+    } catch (InterruptedException e) {
+      TaskInterruptedHelper.throwTaskKillException();
```

Comment:
If `abort` is called due to an exception (`IOException` for ex), this will result in suppressing it.

Reply:
Oh wait, we do need abort even with `IOException`, right ?
Just that we dont want to throw task killed exception - but throw the IOException instead ?

Reply:
> we do need abort even with `IOException`, right ?

Yes


> so that the IOException already raised will propagate

Now change to record the last IOException, call the abort method, and finally throw IOException.